docs/reference/how-to/indexing-speed.asciidoc

```diff
@@ -0,0 +1,95 @@
+[[tune-for-indexing-speed]]
+== Tune for indexing speed
+
+=== Use bulk requests
+
+Bulk requests will yield much better performance than single-document index
+requests. In order to know the optimal size of a bulk request, you shoud run
+a benchmark on a single node with a single shard. First try to index 100
+documents at once, then 200, then 400, etc. doubling the number of documents
+in a bulk request in every benchmark run. When the indexing speed starts to
+plateau then you know you reached the optimal size of a bulk request for your
+data. In case of tie, it is better to err in the direction of too few rather
+than too many documents. Beware that too large bulk requests might put the
+cluster under memory pressure when many of them are sent concurrently, so
+it is advisable to avoid going beyond a couple tens of megabytes per request
+even if larger requests seem to perform better.
+
+=== Increase the refresh interval
+
+The default <<dynamic-index-settings,`index.refresh_interval`>> is `1s`, which
+forces elasticsearch to flush the indexing memory buffer every second.
+Increasing this value (to say, `30s`) will allow larger segments to flush and
+decreases future merge pressure.
+
+=== Disable refresh and replicas for initial loads
+
+If you need to load a large amount of data at once, you should disable refresh
+by setting `index.refresh_interval` to `-1` and set `index.number_of_replicas`
+to `0`. This will temporarily put your index at risk since the loss of any shard
+will cause data loss, but at the same time indexing will be faster since
+documents will be indexed only once. Once the initial loading is finished, you
+can set `index.refresh_interval` and `index.number_of_replicas` back to their
+original values.
+
+=== Use multiple workers/threads to send data to elasticsearch
+
+A single thread sending bulk requests is unlikely to be able to max out the
```

Comment:
I asked @jpountz to put it in the order that I think gives you the biggest bang for the least effort. First you do bulk because you are going to get garbage performance without it. Even if it is more effort. Then you do refresh interval because it is easy to change and buys you a pretty substantial performance bump. I saw 30% the last time I was playing with it but that was a silly external laptop hard drive. Then you do multiple writers because that helps but it is more work than refresh interval.